demo-circuit/src/blaze_csw/constraints/mod.rs

```diff
@@ -0,0 +1,1652 @@
+            "1089ba2f1bee0bbc8f2270541bb22595026fe7d828033845d5ed82f31386b65d",
+            "305510ff60436930d09ccb8e2321211967aadfe904f30ccb13f600786f9e297a",
+            "c80155e642065ca1cc575f69fa658f837b880df76771a335f40ce27240735443",
+            "50e8a8b680918c1840bedfa1650e53f94c8823e81f6efd24d9e37fedfab9344f",
+            "98b4a1c05a44708014a895d27923c7c20f3260e1bc9f2d5edcd6996e4d017944",
+            "b840b87072d095849d433ec11ddd49b138f1823dae16268dcbe46d8035635e74",
+            "b07b2199ad9258449889686423a3c9382cf428355ac348bce40c9d639edf6759",
+            "302fcad55ae4b8f54ab3ab01eaf171873d38676075dff601e4b12a377c7c217d",
+            "001d2489d7b8caab450822ee6393d0b9324da8af67fda2b2cba19b46f64de852",
+            "3811067e9f19d35b2f7487eeb08076a9c4a459dec10791095ebae03bb613f375",
+        ];
+
+        let test_sc_te_public_keys = vec![
+            "f165e1e5f7c290e52f2edef3fbab60cbae74bfd3274f8e5ee1de3345c954a166",
+            "8f80338eef733ec67c601349c4a8251393b28deb722cfd0a91907744a26d3dab",
+            "cc1983469486418cd66dcdc8664677c263487b736840cfd1532e144386fa7610",
+            "88166617f91bc145b243c2ae6e1088f1208bf17311cca74dbf032fee25b219e0",
+            "6f97404947a00311785785217b1759b002cbae16da26e0801f0dcbe4e00d5f45",
+            "fb7a8589cbe59427b2e9c91a5091bf43cf2080f1d4f1947af0d214ca825076f0",
+            "30da57cda802def8dfd764812f2e3c82eb2871b2a14e3bb634f2195ef733796d",
+            "622c8cb09b558fecfc60ce1ec4b1e3014fe04f4628e06cad58ce9ded4d192a2d",
+            "3733056f59780d2f17adf073582634940c6ae57d530345d28e9b6b7cf1d3dcfb",
+            "423cb2cdd87b3e612517cf77e68d918914b0705d8937ef7e25b24a53620bc9d1",
+            "f5206f3569998819efc57e83e8521110e9414c8dca8c5e96c173366e9acd958f",
+            "f1785d4d2f6017ad7a25f795db5beb48d38d6f8cd44dcc3b7f321b8e2a5352fd",
+        ];
+
+        let rng = &mut thread_rng();
+        let random_idx = rng.gen_range(0..test_sc_secrets.len());
+
+        let (test_sc_secret, test_sc_public_key) = (
+            test_sc_secrets[random_idx],
+            test_sc_te_public_keys[random_idx],
+        );
+
+        // Parse pk LE bits
+        let pk_bytes = hex::decode(test_sc_public_key).unwrap();
+
+        // Parse sk LE bits
+        let sk_bytes = hex::decode(test_sc_secret).unwrap();
+        let sk = deserialize_fe_unchecked(sk_bytes.to_vec());
+
+        // Convert it to bits and reverse them (circuit expects them in LE but write_bits outputs in BE)
+        let mut sk_bits = sk.write_bits();
+        sk_bits.reverse();
+
+        (pk_bytes.try_into().unwrap(), sk_bits.try_into().unwrap())
+    }
+
+    fn compute_mst_tree_data(
+        utxo_output_data: CswUtxoOutputData,
+    ) -> (FieldElement, FieldElement, GingerMHTBinaryPath) {
+        let mut mst = GingerMHT::init(MST_MERKLE_TREE_HEIGHT, 1 << MST_MERKLE_TREE_HEIGHT).unwrap();
+
+        let mst_leaf_hash = utxo_output_data.hash(None).unwrap();
+
+        mst.append(mst_leaf_hash).unwrap();
```

Comment:
I see we are not enforcing in the circuit (and thus not computing here) that the position of the leaf (leaf_index) should be equal to the hash of the output_data mod the total number of leaves of the tree. Daniele told me that the design is a bit different from the original, so I guess it is not needed here, but please double check this.

Reply:
For now should be fine, as we don't use the bit vector